### DISCUSSION TITLE
Disable retrospective edits

### DISCUSSION THREAD

**Discussion by tobidwest** [crater-invoice-inc/crater#562] [Ideas]
In some countries, it is mandatory to make sure that accounting records cannot be edited retrospectively. It seems such essential feature is missing in Crater, if I'm not mistaken. I expect a toogle in settings to disable edits once an invoice has been sent or a payment has been recorded. Without this, you aren't allowed to use Crater in many countries.

What would be even better is if there'd be a notes field that's the only one you can edit so you can leave a note if you make a mistake or if Crater would let you edit retrospectively but log all the changes made.

Thanks and stay healthy!

**Comment by Pazekal90** [comment]
You are not mistaken. In Germany the financial government goes very strict with violations.

**Comment by tobidwest** [comment]
It could look like this:
![image](https://user-images.githubusercontent.com/61382793/127646219-ffdf7adf-73f3-4c77-ae11-0d12a78f7068.png)

"Änderungen erlauben" = "Allow edits"

"In some countries it must be ensured that entries in the accounting system cannot be changed afterwards. Activate this option to make changes."

**Comment by tobidwest** [comment]
And when clicking on the edit button, you could show a prompt similar to the one on the updates page:
![image](https://user-images.githubusercontent.com/61382793/127646717-45d2a262-a929-4982-a6b0-4157a509b601.png)

But with this text:
![image](https://user-images.githubusercontent.com/61382793/127646665-f33eba86-24c8-4bdb-a935-05ed73ae965e.png)

"Edits locked"
"You cannot make changes to this record retrospectively"

**Comment by tobidwest** [comment]
What would be even better from a German businessowner's perspective is if the setup assistant would ask wheter you want to lock edits or not and then this setting cannot be changed without manually editing the config file (Invoiceplane handles it this way).

**Comment by mohitpanjwani** [comment]
Good suggestion! We'll add an option on settings to disable editing invoices once the invoice is marked as sent or payment is recorded. 

We can keep following options: 
- Allow Edits (default)
- Disable after payment is recorded
- Disable after invoice is sent

**Comment by tobidwest** [comment]
That'd work, thank you so much!

**Comment by tobidwest** [comment]
Hi @mohitpanjwani, what's the state of affairs? Thanks

**Reply by mohitpanjwani** [reply]
> Hi @mohitpanjwani, what's the state of affairs? Thanks

Hi buddy,

This is already done but it will be released in the major update next month.

**Comment by websplee** [comment]
Following up on this, if you want to "edit" an invoice, is there an option for a credit/debit note?
